scripts/common.py

```diff
@@ -437,6 +468,30 @@ def convert_to_posixpath(file_path):
     return file_path.replace('\\', '/')
 
 
+def inplace_replace_file(filename, regex_pattern, replace):
+    """Replace the file content inplace with regex pattern. The pattern is used
+    to replace the file's content line by line.
+
+    Note:
+        This function was not designed to achive a comprehensive task. Instead,
+        it should only be used with some trivial tasks.
+
+    Args:
+        filename: str. The name of the file to be changed.
+        regex_pattern: str. The pattern to check.
+        replace: str. The content to be replaced.
+    """
+    regex = re.compile(regex_pattern)
+    for line in fileinput.input(
+            files=[filename], inplace=True, backup='.bak'):
+        # As PRINT is going to add a new line symbol, if we do not remove the
```

Comment:
Hm, if this is done just for the PRINT, then we should just do the next three lines as:

```
    python_utils.PRINT(regex.sub(replace, line).rstrip())
```

right?

Also, I'm a bit confused because I can't see where the line gets written back to the file. Could you explain how this works?

Reply:
Aha, I know the `fileinput` API is really confusing, I got the same confusion. Basically all the things sent to stdout will be written back to the file. 

https://docs.python.org/2.7/library/fileinput.html
You can have a look at the `Optional in-place filtering` part.

Reply:
Wow, that _is_ confusing.

Can we just not use it? Let's just do the "boring" thing of: "read the string from file, modify the string, write string back into file". Probably better to emphasize readability/understandability here.

Reply:
Changed